src/lib/port-finder.ts

```diff
@@ -93,6 +106,20 @@ class PortFinder {
 				} );
 		}
 	}
+
+	/**
+	 * Checks if a specific port is available.
+	 *
+	 * @param {number} port - The port number to check.
+	 * @returns {Promise<boolean>} A promise that resolves to true if the port is available, false otherwise.
+	 **/
+	public async isPortAvailable( port: number ): Promise< boolean > {
+		const localPort = this.#searchPort;
+		this.#searchPort = port;
+		const result = await this.#isPortFree();
+		this.#searchPort = localPort;
```

Comment:
This pattern of saving/restoring the global variable `this.#searchPort` seems a bit fiddly to me. What do you think? I would suggest updating the `this.#isPortFree()` to accept a port as a parameter, maybe using `this.#searchPort` as a default, or even better using either `this.#searchPort` as a parameter in the other calls and the `port` in this call.